test/e2e/services.go

```diff
@@ -41,3 +46,42 @@ func newRulesObjstoreService(e e2e.Environment) (*e2e.InstrumentedRunnable, erro
 		},
 	), nil
 }
+
+func newMinio(t *testing.T, e e2e.Environment) {
+	// Create S3 replacement for rules backend
+	bucket := "obs_rules_test"
+	userID := strconv.Itoa(os.Getuid())
+	ports := map[string]int{e2edb.AccessPortName: 8090}
+	envVars := []string{
+		"MINIO_ROOT_USER=" + e2edb.MinioAccessKey,
+		"MINIO_ROOT_PASSWORD=" + e2edb.MinioSecretKey,
+		"MINIO_BROWSER=" + "off",
+		"ENABLE_HTTPS=" + "0",
+		// https://docs.min.io/docs/minio-kms-quickstart-guide.html
+		"MINIO_KMS_KES_ENDPOINT=" + "https://play.min.io:7373",
+		"MINIO_KMS_KES_KEY_FILE=" + "root.key",
+		"MINIO_KMS_KES_CERT_FILE=" + "root.cert",
+		"MINIO_KMS_KES_KEY_NAME=" + "my-minio-key",
+	}
+	f := e2e.NewInstrumentedRunnable(e, "rules-minio", ports, e2edb.AccessPortName)
+	runnable := f.Init(
+		e2e.StartOptions{
+			Image: "minio/minio:RELEASE.2022-03-03T21-21-16Z",
+			// Create the required bucket before starting minio.
+			Command: e2e.NewCommandWithoutEntrypoint("sh", "-c", fmt.Sprintf(
+				// Hacky: Create user that matches ID with host ID to be able to remove .minio.sys details on the start.
+				// Proper solution would be to contribute/create our own minio image which is non root.
+				"useradd -G root -u %v me && mkdir -p %s && chown -R me %s &&"+
+					"curl -sSL --tlsv1.2 -O 'https://raw.githubusercontent.com/minio/kes/master/root.key' -O 'https://raw.githubusercontent.com/minio/kes/master/root.cert' && "+
```

Comment:
just curious, why do we have to downgrade the TLS version?

Reply:
I think it was taken from here: https://docs.min.io/docs/minio-kms-quickstart-guide.html in https://github.com/observatorium/api/pull/251. MinIO seems to only support tlsv1.2 as documented [here](https://docs.min.io/minio/baremetal/security/network-encryption/minio-tls.html).

Reply:
Hmm I'm not sure I agree. The way I read the text from the link you shared is the opposite:
>MinIO uses only supported (non-deprecated) TLS protocols (TLS 1.2 and later).

Minio seems to support everything after and including 1.2, i.e. including 1.3.

Reply:
Yeah, was just about to edit this but you beat me to it! :)
Let me bump it to v1.3!